tests/integration/conftest.py

```diff
@@ -29,7 +29,6 @@ def __hash__(self) -> int:
         return hash(
             (
                 self.__class__,
-                self.data_source_config,
```

Comment:
This hashing function started failing once I added the read/write options to the Pandas/Spark configs.

Reply:
What was the failure? If we don't include this in the hash, I would expect to run into issues with thinking that 2 TestConfigs with different data source types are equivalent. I'm going to throw a blocker on until we've discussed this.

Reply:
Something like "unable to hash dictionary". Wouldn't the hash of `data` always be unique?

Reply:
The original thinking was that when we hash, it should be unique to the data AND the datasource. Like we want to differentiate between MY_DATA for postgres vs MY_DATA for snowflake, but I'm realizing now that this is fine; even on hash collisions, we still store both in the set and it should work. FWIW, I think excluding the new read/write fields from the hash might be the most ideal state (I think you can just pass in `hash=False` to those `field`s. But I think we're fine either way.

Reply:
I'll try the `hash=False` solution and then if that doesn't work I'll try to use `json.dumps`.

Reply:
I trust your judgement on it, but I'd probably use what you currently have over using json.dumps, in case equivalent data serialized differently, e.g. keys in a dictionary showing up in a different order. Maybe we'd never hit that since order is preserved with python 3, but idk. Either way, I think if it works, we should ship it!

Reply:
I do think you can do something like `json.dumps(d, sort_keys=True)` when your keys are all strings.